lighthouse-core/audits/autocomplete.js

```diff
@@ -25,10 +25,152 @@ const UIStrings = {
+  'ADDRESS_BILLING_APT_NUM': 'billing address-level3',
+  'ADDRESS_BILLING_CITY': 'billing address-level2',
+  'ADDRESS_BILLING_STATE': 'billing address-level1',
+  'ADDRESS_BILLING_ZIP': 'billing postal-code',
+  'ADDRESS_BILLING_COUNTRY': 'billing country-name',
+  'CREDIT_CARD_NAME_FULL': 'cc-name',
+  'CREDIT_CARD_NAME_FIRST': 'cc-given-name',
+  'CREDIT_CARD_NAME_LAST': 'cc-family-name',
+  'CREDIT_CARD_NUMBER': 'cc-number',
+  'CREDIT_CARD_EXP_MONTH': 'cc-exp-month',
+  'CREDIT_CARD_EXP_2_DIGIT_YEAR': 'cc-exp-year',
+  'CREDIT_CARD_EXP_4_DIGIT_YEAR': 'cc-exp-year',
+  'CREDIT_CARD_EXP_DATE_2_DIGIT_YEAR': 'cc-exp',
+  'CREDIT_CARD_EXP_DATE_4_DIGIT_YEAR': 'cc-exp',
+  'CREDIT_CARD_TYPE': 'cc-type',
+  'CREDIT_CARD_VERIFICATION_CODE': 'cc-csc',
+  'COMPANY_NAME': 'organization',
+  'PASSWORD': 'current-password',
+  'ACCOUNT_CREATION_PASSWORD': 'new-password',
+  'HTML_TYPE_UNSPECIFIED': 'Requires manual review.',
+  'HTML_TYPE_NAME': 'name',
+  'HTML_TYPE_HONORIFIC_PREFIX': 'honorific-prefix',
+  'HTML_TYPE_GIVEN_NAME': 'given-name',
+  'HTML_TYPE_ADDITIONAL_NAME': 'additional-name',
+  'HTML_TYPE_FAMILY_NAME': 'family-name',
+  'HTML_TYPE_ORGANIZATION': 'organization',
+  'HTML_TYPE_STREET_ADDRESS': 'street-address',
+  'HTML_TYPE_ADDRESS_LINE1': 'address-line1',
+  'HTML_TYPE_ADDRESS_LINE2': 'address-line2',
+  'HTML_TYPE_ADDRESS_LINE3': 'address-line3',
+  'HTML_TYPE_ADDRESS_LEVEL1': 'address-level1',
+  'HTML_TYPE_ADDRESS_LEVEL2': 'address-level2',
+  'HTML_TYPE_ADDRESS_LEVEL3': 'address-level3',
+  'HTML_TYPE_COUNTRY_CODE': 'tel-country-code',
+  'HTML_TYPE_COUNTRY_NAME': 'country-name',
+  'HTML_TYPE_POSTAL_CODE': 'postal-code',
+  'HTML_TYPE_FULL_ADDRESS': 'street-address',
+  'HTML_TYPE_CREDIT_CARD_NAME_FULL': 'cc-name',
+  'HTML_TYPE_CREDIT_CARD_NAME_FIRST': 'cc-given-name',
+  'HTML_TYPE_CREDIT_CARD_NAME_LAST': 'cc-family-name',
+  'HTML_TYPE_CREDIT_CARD_NUMBER': 'cc-number',
+  'HTML_TYPE_CREDIT_CARD_EXP': 'cc-exp',
+  'HTML_TYPE_CREDIT_CARD_EXP_MONTH': 'cc-exp-month',
+  'HTML_TYPE_CREDIT_CARD_EXP_YEAR': 'cc-exp-year',
+  'HTML_TYPE_CREDIT_CARD_VERIFICATION_CODE': 'cc-csc',
+  'HTML_TYPE_CREDIT_CARD_TYPE': 'cc-csc',
+  'HTML_TYPE_TEL': 'tel',
+  'HTML_TYPE_TEL_COUNTRY_CODE': 'tel-country-code',
+  'HTML_TYPE_TEL_NATIONAL': 'tel-national',
+  'HTML_TYPE_TEL_AREA_CODE': 'tel-area-code',
+  'HTML_TYPE_TEL_LOCAL': 'tel-local',
+  'HTML_TYPE_TEL_LOCAL_PREFIX': 'tel-local-prefix',
+  'HTML_TYPE_TEL_LOCAL_SUFFIX': 'tel-local-suffix',
+  'HTML_TYPE_TEL_EXTENSION': 'tel-extension',
+  'HTML_TYPE_EMAIL': 'email',
+  'HTML_TYPE_ADDITIONAL_NAME_INITIAL': 'additional-name-initial',
+  'HTML_TYPE_CREDIT_CARD_EXP_DATE_2_DIGIT_YEAR': 'cc-exp-year',
+  'HTML_TYPE_CREDIT_CARD_EXP_DATE_4_DIGIT_YEAR': 'cc-exp-year',
+  'HTML_TYPE_CREDIT_CARD_EXP_2_DIGIT_YEAR': 'cc-exp-year',
+  'HTML_TYPE_CREDIT_CARD_EXP_4_DIGIT_YEAR': 'cc-exp-year',
+  'HTML_TYPE_UPI_VPA': 'Requires manual review.',
+  'HTML_TYPE_ONE_TIME_CODE': 'one-time-code',
```

Comment:
copy

Reply:
looked into this and even on the same form I see basic types and the "HTML_TYPE_" prefixed ones.  Didn't expect that...


![image](https://user-images.githubusercontent.com/39191/91750754-792fcc80-eb78-11ea-808e-ee33981fb0c8.png)



looking around in the chromium source.... 

* we have this list: https://source.chromium.org/chromium/chromium/src/+/master:components/autofill/core/browser/form_structure.cc;l=173-380;drc=c15160ea3163d6051dc8c042cbecc5a29b770da6 how the primary tokens map to the "HTML_TYPE_*" equivalents. (we more or less want the inverse mapping)
* and then we also have this https://source.chromium.org/chromium/chromium/src/+/master:components/autofill/core/browser/autofill_type.cc;l=397;drc=6e8b402a6231405b753919029c9027404325ea00 which maps "HTML_TYPE_" types back to basic ones.

anyhow... at this point i'm not entirely sure what sort of values the prediction can have... but the `autofillSuggestions` mapping that George compiled looks comprehensive enough to me. It's possible it may be too inclusive, but we need some questions answered first. I've reach out to the autofill folks in Slack with questions. In the meantime, this mapping lgtm.